src/util-lua-common.c

```diff
@@ -715,42 +715,41 @@ static int LuaCallbackLogError(lua_State *luastate)
  */
 static int LuaCallbackFileInfoPushToStackFromFile(lua_State *luastate, const File *file)
 {
-#ifdef HAVE_NSS
-    char md5[33] = "";
-    char *md5ptr = md5;
-    if (file->flags & FILE_MD5) {
-        size_t x;
-        for (x = 0; x < sizeof(file->md5); x++) {
-            char one[3] = "";
-            snprintf(one, sizeof(one), "%02x", file->md5[x]);
-            strlcat(md5, one, sizeof(md5));
+    char *md5ptr = NULL;
+    char *sha1ptr = NULL;
+    char *sha256ptr = NULL;
+    if (!g_disable_hashing) {
+        char md5[33] = "";
+        md5ptr = md5;
+        if (file->flags & FILE_MD5) {
+            size_t x;
+            for (x = 0; x < sizeof(file->md5); x++) {
```

Comment:
Can we use a `ToHex` func here as well?

Reply:
No, not without a larger refactor.  The checksum has already been finalized, so the context is gone.  We also have the same issue in the fileinfo logging.

Same argument applies here to the email body md5..  We could keep the context alive by not calling finalize.  Then each time we log it we clone it, then finalize to hex.

Computationally it seems to make more sense to store the hex buffer once and reuse it, like in the email. But that leads to struct growth. The alternative leads to more CPU usage as the checksum is potentially cloned, finalized and formatted multiple times.